src/lib/OS/SimOS.cc

```diff
@@ -270,5 +321,30 @@ VAddrTranslator SimOS::getVAddrTranslator() {
   return fn;
 }
 
+void SimOS::createSpecialFileDirectory() const {
+  simeng::SpecialFileDirGen SFdir = simeng::SpecialFileDirGen();
+  // Remove any current special files dir
+  SFdir.RemoveExistingSFDir();
+  // Create new special files dir
+  SFdir.GenerateSFDir();
+}
+
+void SimOS::terminateThreadHelper(std::shared_ptr<Process> proc) {
+  uint64_t tid = proc->getTID();
+  if (proc->status_ == procStatus::executing) {
+    // Set core's status to idle, stopping execution immediately
+    for (auto core : cores_) {
+      if (core->getCurrentTID() == tid) {
+        core->setStatus(CoreStatus::idle);
+        break;
+      }
+    }
+  } else {
```

Comment:
Couldn't we just set the status of the process to be complete in all cases? I'm aware that the process doesn't live in either queue if it's executing but just thinking about avoiding any future error with different schedulers. Not too attached to the idea so don't mind if we keep the current logic.

Reply:
Changed so a process in any state is updated to `completed`